### PR TITLE
Add lowerings and module implementation for TensorList concat

### PR DESCRIPTION
**Pull request by rsuderman**
Handles each part of the tensorlist lowering from TF_TensorListConcatV2Op down the to the tensorlist dialect, with execution via the IREE tensorlist module.

Includes tests for each step of the lowering process.

### CODE REVIEW

File: integrations/tensorflow/compiler/dialect/tf_tensorlist/conversion/convert_tf_to_tf_tensorlist.cc [hunk 1]
```diff
@@ -37,6 +37,33 @@ bool isTfVariant(Type type) {
...
 
+namespace {
+class ConvertTfTensorlistConcatV2
+    : public OpRewritePattern<TF::TensorListConcatV2Op> {
+ public:
+  using OpRewritePattern<TF::TensorListConcatV2Op>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(TF::TensorListConcatV2Op op,
+                                PatternRewriter &rewriter) const override {
+    llvm::errs() << "Found op\n";
```
Comment (silvasean) [inline integrations/tensorflow/compiler/dialect/tf_tensorlist/conversion/convert_tf_to_tf_tensorlist.cc:48]: remove debug code

File: integrations/tensorflow/compiler/dialect/tf_tensorlist/conversion/convert_tf_to_tf_tensorlist.cc [hunk 2]
```diff
@@ -37,6 +37,33 @@ bool isTfVariant(Type type) {
...
+    op.dump();
+    Value tensor_lists = op.input_handle();
```
Comment (silvasean) [inline integrations/tensorflow/compiler/dialect/tf_tensorlist/conversion/convert_tf_to_tf_tensorlist.cc:50]: why is tensor_lists plural?

File: integrations/tensorflow/compiler/dialect/tf_tensorlist/ir/tf_tensorlist_ops.td [hunk 3]
```diff
@@ -138,4 +138,45 @@ def TfTensorList_Stack : TfTensorList_Op<"Stack"> {
...
+  let arguments = (ins
+    TfTensorList_TensorList:$list
+  );
+
+  let results = (outs
+    TF_Tensor:$tensor
+  );
+}
+
+// TODO(suderman): This op purely exists to be optimized away right now.
```
Comment (silvasean) [inline integrations/tensorflow/compiler/dialect/tf_tensorlist/ir/tf_tensorlist_ops.td:160]: This is used by the gradient code. Maybe rewrite this as "if gradients are not being calculated, the value will be DCE'd"

File: iree/compiler/Dialect/Modules/TensorList/Conversion/test/convert_hal_to_vm.mlir [hunk 4]
```diff
@@ -27,3 +27,21 @@ func @SetItem(%list: !tensorlist.list, %index: !hal.buffer_view, %item: !hal.buf
   return %0 : !tensorlist.list
 }
 // CHECK: vm.import @tensorlist.set_item
+
+// -----
+
+// CHECK-LABEL: @Stack
+func @Stack(%list: !tensorlist.list, %element_shape: !hal.buffer_view, %num_elements: !hal.buffer_view) -> !hal.buffer_view {
```
Comment (silvasean) [inline iree/compiler/Dialect/Modules/TensorList/Conversion/test/convert_hal_to_vm.mlir:34]: Thanks for agumenting the test coverage for Stack in this CL :)))

File: iree/compiler/Dialect/Modules/TensorList/IR/TensorListOps.td [hunk 5]
```diff
@@ -111,4 +111,22 @@ def TensorList_Stack : Op<TensorList_Dialect, "Stack"> {
...
 
+def TensorList_Concat : Op<TensorList_Dialect, "Concat"> {
+  let summary = "Creates a tensor by concatenate the tensors in the tensorlist";
+  let description = [{
+    Creates a new tensor `tensor` by concatenating the tensors in `list` along
+    the leading dimension.
+
+    Requires the list to be non-empty.
+    Requires all tensors contained in `list` to have the same dimensions along
+    the non-leading axis.
```
Comment (silvasean) [inline iree/compiler/Dialect/Modules/TensorList/IR/TensorListOps.td:122]: axis -> axes

File: integrations/tensorflow/compiler/dialect/tf_tensorlist/ir/tf_tensorlist_ops.td (hunk 3 above)
Comment (rsuderman) [reply integrations/tensorflow/compiler/dialect/tf_tensorlist/ir/tf_tensorlist_ops.td:160]: Done.

File: iree/compiler/Dialect/Modules/TensorList/Conversion/test/convert_hal_to_vm.mlir (hunk 4 above)
Comment (rsuderman) [reply iree/compiler/Dialect/Modules/TensorList/Conversion/test/convert_hal_to_vm.mlir:34]: No problem :P. Was an easy drive by.

File: iree/compiler/Dialect/Modules/TensorList/IR/TensorListOps.td (hunk 5 above)
Comment (rsuderman) [reply iree/compiler/Dialect/Modules/TensorList/IR/TensorListOps.td:122]: Done.

File: integrations/tensorflow/compiler/dialect/tf_tensorlist/conversion/convert_tf_to_tf_tensorlist.cc (hunk 2 above)
Comment (rsuderman) [reply integrations/tensorflow/compiler/dialect/tf_tensorlist/conversion/convert_tf_to_tf_tensorlist.cc:48]: No reason. Renamed.

**Comment by rsuderman** [comment]
> remove debug code 

Removed

**Pull request closed without merging**